test/tests_basic.jl

```diff
@@ -74,6 +102,84 @@ end
     @test_nowarn p4est_destroy(p4est)
     @test_nowarn p4est_connectivity_destroy(connectivity)
   end
+
+  @testset "p8est_connectivity_new_unitcube" begin
+    connectivity = @test_nowarn p8est_connectivity_new_unitcube()
+    @test_nowarn p8est_connectivity_destroy(connectivity)
+  end
+
+  @testset "p4est_refine_coarsen" begin
+    connectivity = @test_nowarn p4est_connectivity_new_periodic()
+    p4est = @test_nowarn p4est_new(MPI.COMM_WORLD, connectivity, 0, C_NULL, C_NULL)
+
+    refine_fn_c = @cfunction(refine_fn, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+    @test_nowarn p4est_refine(p4est, true, refine_fn_c, C_NULL)
+    coarsen_fn_c = @cfunction(coarsen_fn, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+    @test_nowarn p4est_coarsen(p4est, true, coarsen_fn_c, C_NULL)
+    @test_nowarn p4est_destroy(p4est)
+    @test_nowarn p4est_connectivity_destroy(connectivity)
+  end
+
+  @testset "local_num_quadrants" begin
+    connectivity = @test_nowarn p4est_connectivity_new_periodic()
+    p4est = @test_nowarn p4est_new(MPI.COMM_WORLD, connectivity, 0, C_NULL, C_NULL)
+    @test_nowarn Int(unsafe_load(p4est).local_num_quadrants)
+  end
+
+  # This test is inspired by test/test_balance_type2.c from p4est (only for 2D)
+  @testset "p4est_balance" begin
+    connectivity = @test_nowarn p4est_connectivity_new_star()
+    p4est = @test_nowarn p4est_new_ext(MPI.COMM_WORLD, connectivity, 0, 0, 0, 0, C_NULL, C_NULL)
+  refine_fn_balance_c = @cfunction(refine_fn_balance, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+    @test_nowarn p4est_refine(p4est, 1, refine_fn_balance_c, C_NULL)
+    # face balance
+    p4estF = @test_nowarn p4est_copy(p4est, 0)
+    @test_nowarn p4est_balance(p4estF, P4EST_CONNECT_FACE, C_NULL)
+    crcF = @test_nowarn p4est_checksum(p4estF)
+    println("Face balance with ", unsafe_load(p4estF).global_num_quadrants, " quadrants and crc ", crcF)
```

Comment:
Would it make sense to test the global number of quadrants here instead of only printing it?